app/queries/get_content_collection.rb

```diff
@@ -36,6 +40,10 @@ def content_items
 
       live_items = live_items.where(publishing_app: @publishing_app) if @publishing_app.present?
 
+      @draft_versions = Version.in_bulk(draft_items, DraftContentItem)
```

Comment:
That would indeed make things quite a lot simpler. See the comment to 3c332f3 though - we're being quite careful to keep Versions and ContentItems separate, at least for now.